sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -362,7 +362,9 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
       str
     }
     logWarning(s"Codegen disabled for this expression:\n $logMessage")
-    InterpretedPredicate.create(expression, inputSchema)
+    val predicate = InterpretedPredicate.create(expression, inputSchema)
+    predicate.initialize(0)
+    predicate
```

Comment:
The partition index is not always 0.

Reply:
I checked the few places calling `newPredicate`. They are either already do `initialize` properly, or don't need to do it.

Reply:
thanks, remove it.